packages/babel-plugin/src/css-prop/__tests__/behaviour.test.tsx

```diff
@@ -390,4 +390,101 @@ describe('css prop behaviour', () => {
 
     expect(actual).toInclude(':hover{color:red}');
   });
+
+  it('should apply conditional logical expression object spread styles', () => {
+    const actual = transform(`
+      import '@compiled/react';
+      import React from 'react';
+
+      const Component = props => (
+        <div css={{
+          ...props.isPrimary && {
+            color: 'blue',
+            fontSize: 20,
+          }
+        }}>hello world</div>
+      );
+    `);
+
+    expect(actual).toInclude('className={ax([props.isPrimary&&"_syaz13q2 _1wybgktf"])}');
+  });
+
+  it('should apply conditional logical expression object styles', () => {
+    const actual = transform(`
+      import '@compiled/react';
+      import React from 'react';
+
+      const Component = props => (
+        <div css={{
+          ':hover': props.isPrimary && {
+            color: 'blue',
+            fontSize: 20,
+          }
+        }}>hello world</div>
+      );
+    `);
+
+    expect(actual).toInclude('className={ax([props.isPrimary&&"_30l313q2 _e915gktf"])}');
+  });
+
+  it('should combine conditional logical expressions', () => {
+    const actual = transform(`
+      import '@compiled/react';
+      import React from 'react';
+
+      const Component = props => (
+        <div css={{
+          ':hover': props.isPrimary && {
+            color: 'blue',
+            ...props.isBold && {
+              fontWeight: 700,
+            },
+          }
+        }}>hello world</div>
+      );
+    `);
+
+    expect(actual).toInclude(
+      'className={ax([props.isPrimary&&"_30l313q2",props.isPrimary&&props.isBold&&"_79b11fw0"])}'
+    );
+  });
+
+  it('should apply multi conditional logical expression', () => {
+    const actual = transform(`
+      import '@compiled/react';
+      import React from 'react';
+
+      const Component = props => (
+        <div css={{
+          ...(props.isPrimary || props.isMaybe) && {
+            color: 'blue',
+            fontSize: 20,
+          }
+        }}>hello world</div>
+      );
+    `);
+
+    expect(actual).toInclude('ax([(props.isPrimary||props.isMaybe)&&"_syaz13q2 _1wybgktf"])');
+  });
+
+  it('should apply array conditional css', () => {
+    const actual = transform(`
+      import '@compiled/react';
+      import React from 'react';
+
+      const Component = props => (
+        <div css={[
+          { fontSize: 40, },
+          (props.isPrimary || props.isMaybe) && {
+            color: 'blue',
+            fontSize: 20,
+          },
+        ]}>hello world</div>
+      );
+    `);
+
+    expect(actual).toInclude(
+      'ax(["_1wyb1ylp",(props.isPrimary||props.isMaybe)&&"_syaz13q2 _1wybgktf"])'
+    );
+  });
```

Comment:
This case is not working:

```ts
const Component = props => (
  <div css={{
    ...props.isPrimary && {
      color: 'blue',
      fontSize: 20,
    },
    border: '1px solid black'
  }}>hello world</div>
);
```

Getting this output:
```ts
/* File generated by @compiled/babel-plugin v0.0.0 */
    
import * as React from 'react';
import { ax, CC, CS } from "@compiled/react/runtime";
const _2 = "._1wybgktf{font-size:20px}";
const _ = "._syaz13q2{color:blue}";

const Component = props => <CC>
    <CS>{[_, _2]}</CS>
    {<div className={ax([props.isPrimary && "_syaz13q2 _1wybgktf"])}>hello world</div>}
  </CC>;
```

but if we put `border: '1px solid black'` before condition `props.isPrimary`, it works

Reply:
nice catch